pkg/manifests/manifests.go

```diff
@@ -1496,6 +1523,40 @@ func (f *Factory) setupQueryLogFile(p *monv1.Prometheus, queryLogFile string) er
 	return nil
 }
 
+// setupProfilesToIgnore configures the label selectors of the Prometheus ("p")
+// to select any ServiceMonitor's or PodMonitor's that doesn't have the scrape
+// profile label or that matches the Scrape Profile ("sp").
+func setupProfilesToIgnore(p *monv1.Prometheus, sp ScrapeProfile) error {
+	// Our goal is to configure Prometheus to select both the resources that
+	// either don't have the scrape profile label or have the desired value.
+	// However with label selectors we are not able to express OR conditions.
+	// Hence, the only alternative is to configure Prometheus to not select any
+	// resource that matches either of the scrape profiles that we are not
+	// interested in.
+	profiles := make([]string, 0, len(ScrapeProfiles)-1)
+	for _, profile := range ScrapeProfiles {
+		if profile == sp {
+			continue
+		}
+		profiles = append(profiles, string(profile))
+	}
+
+	labelSelector := &metav1.LabelSelector{
+		MatchExpressions: []metav1.LabelSelectorRequirement{
+			{
+				Key:      scrapeProfileLabel,
+				Operator: metav1.LabelSelectorOpNotIn,
+				Values:   profiles,
+			},
+		},
+	}
+
+	p.Spec.ServiceMonitorSelector = labelSelector
+	p.Spec.PodMonitorSelector = labelSelector
```

Comment:
I wonder if it shouldn't update p.Spec.ProbeSelector too? This would be mostly for consistency as we're not supporting probes for cluster monitoring anyway.

Reply:
Initially I had it but I removed it because of this comment https://github.com/openshift/cluster-monitoring-operator/pull/1785#discussion_r986555681

Reply:
ah interesting, i'll look into why this has changed (probably no good reason)...

Reply:
hmm my previous comment was completely wrong: I can't find anywhere that probeSpec had been set to nil...